src/database/postgresql/postgresql.js

```diff
@@ -0,0 +1,108 @@
+const { Client } = require('pg');
+const exec = require('./../../utils/await-exec');
+const isGzip = require('./../../utils/isGzip');
+const files = require('./../../utils/files');
+const path = require('path');
+const encryptionCheck = require('../../utils/isEncrypted');
+
+let awaitPostgresqlConnect = (connection) => {
+    return new Promise((resolve, reject) => {
+        connection.connect(function (err) {
+            if (err) {
+                reject(err);
+                return;
+            }
+            resolve(connection);
+        });
+    });
+};
+
+let awaitPostgresqlEnd = (connection) => {
+    return new Promise((resolve, reject) => {
+        connection.end(function (err) {
+            if (err) {
+                reject(err);
+                return;
+            }
+            resolve(connection);
+        });
+    });
+};
+
+let connect = async (dbConfig) => {
+    const connection = new Client({
+        user: dbConfig.dbAuthUser,
+        password: dbConfig.dbAuthPwd,
+        host: dbConfig.dbHost,
+        port: dbConfig.dbPort,
+        database: dbConfig.dbName,
+    });
+
+    const connRes = await awaitPostgresqlConnect(connection);
+    const disConnRes = await awaitPostgresqlEnd(connection);
+
+    return connRes;
+};
+
+let dump = async (dbConfig, key, backupPath) => {
+    const postgresqlDumpCmd = `PGPASSWORD=${dbConfig.dbAuthPwd} \
+    pg_dump --host=${dbConfig.dbHost} \
+    --port=${dbConfig.dbPort} \
+    --username=${dbConfig.dbAuthUser} \
+    --dbname=${dbConfig.dbName} \
+    --compress=0..9 \
+    --format=c
+    > ${backupPath}`;
+
+    let dbDump = await exec(postgresqlDumpCmd);
+
+    if (dbConfig.dbIsCompressionEnabled) {
+        await files.compressFile(backupPath);
+    }
+    if (dbConfig.backupEncryptionEnabled) {
+        await files.encrypt(backupPath, key);
+    }
+    return dbDump;
+};
+
+let restore = async (dbConfig, key, backupFilename) => {
+    let backupFilePath = path.join(dbConfig.dbBackupPath, backupFilename);
+    let decryptFileName;
+    let isEncrypted = await encryptionCheck.isEncrypted(backupFilePath);
+    if (isEncrypted) {
+        await files.decrypt(backupFilePath, key);
+        backupFilePath = `${backupFilePath}_unenc`;
+        decryptFileName = backupFilePath;
+    }
+    let isCompressed = isGzip(backupFilePath);
+    if (isCompressed) {
+        await files.decompressFile(backupFilePath);
+        backupFilePath = `${backupFilePath}.sql`;
+    }
+    const postgresqlRestoreCmd = `PGPASSWORD=${dbConfig.dbAuthPwd} \
```

Comment:
Make this command compatible for powershell(windows) as well.

Reply:
ok